dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -60,18 +74,35 @@ public CommunityRestRepository() {
     protected CommunityRest createAndReturn(Context context) throws AuthorizeException {
         HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
         ObjectMapper mapper = new ObjectMapper();
-        CommunityRest communityRest = null;
+        CommunityRest communityRest;
         try {
             ServletInputStream input = req.getInputStream();
             communityRest = mapper.readValue(input, CommunityRest.class);
         } catch (IOException e1) {
             throw new UnprocessableEntityException("Error parsing request body: " + e1.toString());
         }
 
-        Community community = null;
+        Community community;
+
 
         try {
-            community = cs.create(null, context);
+            Community parent = null;
+            String parentCommunityString = req.getParameter("parent");
+            if (StringUtils.isNotBlank(parentCommunityString)) {
+
+                UUID parentCommunityUuid = UUIDUtils.fromString(parentCommunityString);
+                if (parentCommunityUuid == null) {
+                    throw new BadRequestException("The given parent parameter was invalid: "
+                            + parentCommunityString);
+                }
+
+                parent = cs.find(context, parentCommunityUuid);
+                if (parent == null) {
+                    throw new UnprocessableEntityException("Parent community for id: "
+                            + parentCommunityUuid + " not found");
+                }
+            }
+            community = cs.create(parent, context);
```

Comment:
Just realized that `parent` **may be null** at this point.  Should most/all of this method be included in the `if (StringUtils.isNotBlank(parentCommunityString))`

Reply:
I agree with the comment above and this will be altered to throw a BadRequestException if the parent is left null for the CollectionRestRepository. However to create a top level community, we need to be able to give null to this method as we do not have a CommunityService.create method without a parent parameter. So if we encapsulate this into the surrounding if, we will never be able to create a top level community.
This parent may be left null for the CommunityRestRepository